lib/components/admin/field-trip-windows.js

```diff
@@ -0,0 +1,215 @@
+import moment from 'moment'
+import React, { Component } from 'react'
+import { Badge } from 'react-bootstrap'
+import { connect } from 'react-redux'
+
+import * as callTakerActions from '../../actions/call-taker'
+import DraggableWindow from './draggable-window'
+import FieldTripDetails from './field-trip-details'
+import Icon from '../narrative/icon'
+
+const TABS = [
+  {id: 'new', label: 'New', filter: (req) => req.status !== 'cancelled' && (!req.inboundTripStatus || !req.outboundTripStatus)},
+  {id: 'planned', label: 'Planned', filter: (req) => req.status !== 'cancelled' && req.inboundTripStatus && req.outboundTripStatus},
+  {id: 'cancelled', label: 'Cancelled', filter: (req) => req.status === 'cancelled'},
+  {id: 'past', label: 'Past', filter: (req) => req.travelDate && moment(req.travelDate).diff(moment(), 'days') < 0},
+  {id: 'all', label: 'All', filter: (req) => true}
+]
+
+const SEARCH_FIELDS = [
+  'address',
+  'ccLastFour',
+  'ccName',
+  'ccType',
+  'checkNumber',
+  'city',
+  'classpassId',
+  'emailAddress',
+  'endLocation',
+  'grade',
+  'phoneNumber',
+  'schoolName',
+  'startLocation',
+  'submitterNotes',
+  'teacherName'
+]
+
+/**
+ * Collects the various draggable windows for the Field Trip module.
+ */
+class FieldTripWindows extends Component {
+  _onClickFieldTrip = (request) => {
+    const {callTaker, fetchFieldTripDetails, setActiveFieldTrip} = this.props
+    if (request.id === callTaker.fieldTrip.activeId) {
+      this._onCloseActiveFieldTrip()
+    } else {
+      setActiveFieldTrip(request.id)
+      fetchFieldTripDetails(request.id)
+    }
+  }
+
+  _onCloseActiveFieldTrip = () => {
+    this.props.setActiveFieldTrip(null)
+  }
+
+  _onSearchChange = e => {
+    this.props.setFieldTripFilter({search: e.target.value})
+  }
+
+  _onTabChange = e => {
+    this.props.setFieldTripFilter({tab: e.currentTarget.name})
+  }
+
+  render () {
+    const {callTaker, toggleFieldTrips} = this.props
+    const {fieldTrip} = callTaker
+    const {activeId, filter} = fieldTrip
+    const activeTab = TABS.find(tab => tab.id === filter.tab)
+    const visibleRequests = fieldTrip.requests.data
+      .filter(ft => {
+        if (activeTab) return activeTab.filter(ft)
+        else return true
+      })
```

Comment:
Seems like this could be combined with the following filter function.

Reply:
fixed in 390dede51ba4a08a57ec986de526ced871864f79